### ISSUE TITLE
S_unshare_hek_or_pvn failed assertion

### ISSUE THREAD

**Issue by atoomic** [rurban/perl-compiler#251]
In order to reproduce this bug, you need to create the file, the oneliner will not trigger this issue
Tested with perl 5.22 and debug enabled

````
> cat test.pl
#!./perl

open foo::bar, "test.pl" or die;
*foo:: = *bar::; # gv is now *__ANON__::bar

open thwat::snin, "test.pl" or die;
*thwat:: = *snin::; # gv is now *__ANON__::snin

print qq/ok\n/;
> perl test.pl
ok
> perlcc -r test.pl
test: hv.c:2784: S_unshare_hek_or_pvn: Assertion `he->shared_he_he.hent_hek == hek' failed.
````

**Comment by rurban** [comment]
Not yet fixed with shared_hek_assert=1fa99b0

**Comment by rurban** [comment]
This bug caused me to disable -fcog since 5.10 in release 1.06:
"Had to disable -O1 -fcog (pv_copy_on_grow) on 5.10 and higher until I find out how to fool S_unshare_hek_or_pvn. This fixes all C -O1 and -O2 tests. Warn about this."
Fixed now with 95d488a

**Comment by toddr** [comment]
#255 appears to have fixed this case.
